src/test/java/org/pesmypetcare/webservice/service/MedicationServiceTest.java

```diff
@@ -0,0 +1,152 @@
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getMedicationData(any(String.class), any(String.class),
+                    isA(String.class));
+            service.getMedicationData(owner, petName, dateName);
+        }, "Should return an exception when retrieving a Medication fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetSetOfMedicationsRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getAllMedicationData(owner, petName)).thenReturn(MedicationList);
+        List<Map<List<String>, Object>> list = service.getAllMedicationData(owner, petName);
+        assertSame(MedicationList, list, "Should return a list of Medication entities");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetSetOfMedicationsRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getAllMedicationData(any(String.class), isA(String.class));
+            service.getAllMedicationData(owner, petName);
+        }, "Should return an exception when retrieving a set of Medications fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetMedicationsBetweenDatesRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getAllMedicationsBetween(owner, petName, date, date2)).thenReturn(MedicationList);
+        List<Map<List<String>, Object>> list = service.getAllMedicationsBetween(owner, petName, date, date2);
+        assertSame(MedicationList, list, "Should return a list of Medication entities");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMedicationsBetweenDatesRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getAllMedicationsBetween(any(String.class),
+                    isA(String.class), isA(String.class), isA(String.class));
+            service.getAllMedicationsBetween(owner, petName, date, date2);
+        }, "Should return an exception when retrieving a set of Medications between dates fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationFieldWhenMedicationFieldRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getMedicationField(owner, petName, date, field)).thenReturn(value);
+        Object obtainedValue = service.getMedicationField(owner, petName, date, field);
+        assertSame(value, obtainedValue, "Should return an Object");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMedicationFieldRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getMedicationField(any(String.class), any(String.class),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 127).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)